housekeeper/cli/delete.py

```diff
@@ -99,54 +101,79 @@ def version_cmd(context, bundle_name, version_id, yes):
 @click.pass_context
 def files_cmd(context, yes, tag, bundle_name, before, notondisk, list_files, list_files_verbose):
     """Delete files based on tags."""
+
+    validate_delete_options(tag=tag, bundle_name=bundle_name)
+    before_date = parse_date(before) if before else None
     store = context.obj["store"]
-    file_objs = []
-    if not (tag or bundle_name):
-        LOG.info("Please specify a bundle or a tag")
-        raise click.Abort
 
     if bundle_name:
-        bundle_obj = store.bundle(bundle_name)
-        if bundle_obj is None:
-            LOG.warning("Bundle not found")
-            raise click.Abort
+        validate_bundle_exists(store, bundle_name)
 
-    query = store.files_before(bundle=bundle_name, tags=tag, before=before)
+    files = store.get_files_before(bundle=bundle_name, tags=tag, before_date=before_date)
 
     if notondisk:
-        file_objs = set(query) - store.files_ondisk(query)
-    else:
-        file_objs = query.all()
-
-    nr_files = 0
-    for nr_files, file_obj in enumerate(file_objs, 1):
-        if list_files_verbose:
-            tags = ", ".join(tag.name for tag in file_obj.tags)
-            click.echo(
-                f"{click.style(str(file_obj.id), fg='blue')} | {file_obj.full_path} | "
-                f"{click.style(tags, fg='yellow')}"
-            )
-        elif list_files:
-            click.echo(file_obj.full_path)
-        else:
-            continue
-
-    if nr_files == 0:
-        LOG.warning("no files found")
+        files = store.get_files_not_on_disk(files)
+
+    if not files:
+        LOG.warning("No files found")
         raise click.Abort
 
-    if not (yes or click.confirm(f"Are you sure you want to delete {len(file_objs)} files?")):
+    if list_files_verbose:
+        list_files_verbosely(files)
+
+    elif list_files:
+        list_files_with_full_path(files)
```

Comment:
named args, possibly

```suggestion
        list_files_with_full_path(files=files)
```